File: ExcelDivider/src/main/java/ru/tsystems/divider/exceptions/PropertiesException.java
package ru.tsystems.divider.exceptions;

public class PropertiesException extends Exception {

    /**
     * Constructs a new exception with {@code null} as its detail message.
     * The cause is not initialized, and may subsequently be initialized by a
     * call to {@link #initCause}.
     */
    public PropertiesException(String propertyName, String comment) {
        super("Wrong property! Property name: " + propertyName + "; " + comment);
    }
}

File: ExcelDivider/src/main/kotlin/ru/tsystems/divider/exceptions/PropertiesException.kt
package ru.tsystems.divider.exceptions

class PropertiesException : Exception {

    /**
     * Constructs a new exception with `null` as its detail message.
     * The cause is not initialized, and may subsequently be initialized by a
     * call to [.initCause].
     */
    constructor(
        propertyName: String,
        comment: String
    ) : super("Wrong property! Property name: $propertyName; $comment") {
    }

    constructor(property: String) : super("No obligatory property: $property")
}